tools/cibadmin.c

```diff
@@ -94,7 +93,9 @@ print_xml_output(xmlNode * xml)
         }
 
     } else {
-        pcmk__xml2fd(STDOUT_FILENO, xml);
+        char *buffer = dump_xml_formatted(xml);
```

Comment:
`pcmk__xml2text()` avoids an unnecessary memory allocation in all these places.  The `dump_xml_*()` functions allocate a GString, copy it into a C string, free the GString, and return the C string.

`dump_xml_formatted()` is easier to read and is the pre-c63a898 apprach, and performance isn't important in CLI tools, so I don't have a strong preference.